plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/BridgingHiveMetastore.java

```diff
@@ -154,6 +158,52 @@ public List<String> getAllViews(String databaseName)
         return delegate.getAllViews(databaseName);
     }
 
+    @Override
+    public Map<String, List<String>> getAllTables()
+    {
+        checkState(supportBatchListingOperations(), "Batch listing operations not supported");
+        return delegate.getAllTables();
+    }
+
+    @Override
+    public Map<String, List<String>> getAllViews()
+    {
+        checkState(supportBatchListingOperations(), "Batch listing operations not supported");
+        return delegate.getAllViews();
+    }
+
+    @Override
+    public boolean supportBatchListingOperations()
+    {
+        return switch (batchListingOperationsSupport) {
+            case UNKNOWN -> {
+                if (!delegate.supportBatchGetViews()) {
+                    batchListingOperationsSupport = BatchListingOperationsSupport.UNSUPPORTED;
+                }
+                else {
+                    batchListingOperationsSupport = checkBatchListingOperationsSupport();
+                }
+                yield supportBatchListingOperations();
+            }
+            case SUPPORTED -> true;
+            case UNSUPPORTED -> false;
+        };
+    }
+
+    private BatchListingOperationsSupport checkBatchListingOperationsSupport()
+    {
+        try {
+            delegate.getAllTables();
+            return BatchListingOperationsSupport.SUPPORTED;
+        }
+        catch (TrinoException e) {
+            if (e.getErrorCode() == HIVE_METASTORE_ERROR.toErrorCode()) {
+                return BatchListingOperationsSupport.UNSUPPORTED;
```

Comment:
What is HMS is (still) down during the call to getAllTables and we (incorrect) remember that batching cannot be used?

Reply:
That is a valid argument I'm afraid.
I see 2 reasonable options:
1.We invoke getAll* explicitly from outside, i.e. HiveMetadata.listTables method and keep a separate cache for "all-tables" (and all views). This is more straightforward and object-oriented. However, ew are keeping the data doubled in cache. The solution is not transparent to the invoker though. And invoking `getTable(dbName)` after `getAllTables` will result in two call to metastore. Now it is one.
2. We initialize these caches lazily, only after the first get, i.e. when its not initialize we perform a dummy getAllTables and see whether metastore allows it and then decide what type of cache to init.